src/serac/physics/utilities/functional/domain_integral.hpp

```diff
@@ -590,19 +495,44 @@ class DomainIntegral {
     //
     // note: the qf_derivatives_ptr is copied by value to each lambda function below,
     //       to allow the evaluation kernel to pass derivative values to the gradient kernel
-    evaluation_ = [=, &data](const mfem::Vector& U, mfem::Vector& R) {
-      domain_integral::evaluation_kernel<geometry, test_space, trial_space, Q>(U, R, qf_derivatives.get(), J_, X_,
-                                                                               num_elements, qf, data);
-    };
-
-    gradient_ = [=](const mfem::Vector& dU, mfem::Vector& dR) {
-      domain_integral::gradient_kernel<geometry, test_space, trial_space, Q>(dU, dR, qf_derivatives.get(), J_,
-                                                                             num_elements);
-    };
-
-    gradient_mat_ = [=](mfem::Vector& K_e) {
-      gradient_matrix_kernel<geometry, test_space, trial_space, Q>(K_e, qf_derivatives.get(), J_, num_elements);
-    };
+
+    if constexpr (std::is_same_v<execution_policy, serac::cpu_policy>) {
+      evaluation_ = [=, &data](const mfem::Vector& U, mfem::Vector& R) {
+        domain_integral::evaluation_kernel<geometry, test_space, trial_space, Q>(U, R, qf_derivatives.get(), J_, X_,
+                                                                                 num_elements, qf, data);
+      };
+
+      gradient_ = [=](const mfem::Vector& dU, mfem::Vector& dR) {
+        domain_integral::gradient_kernel<geometry, test_space, trial_space, Q>(dU, dR, qf_derivatives.get(), J_,
+                                                                               num_elements);
+      };
+
+      gradient_mat_ = [=](mfem::Vector& K_e) {
+        gradient_matrix_kernel<geometry, test_space, trial_space, Q>(K_e, qf_derivatives.get(), J_, num_elements);
+      };
+    }
+
+    if constexpr (std::is_same_v<execution_policy, serac::gpu_policy>) {
+#if defined(__CUDACC__)
+      evaluation_ = [=](const mfem::Vector& U, mfem::Vector& R) {
+        serac::detail::ThreadExecutionConfiguration exec_config{.blocksize = 128};
+
+        domain_integral::evaluation_kernel_cuda<geometry, test_space, trial_space, Q,
+                                                serac::detail::ThreadExecutionPolicy::THREAD_PER_QUADRATURE_POINT>(
+            exec_config, U, R, qf_derivatives.get(), J_, X_, num_elements, qf);
+      };
+
+      gradient_ = [=](const mfem::Vector& dU, mfem::Vector& dR) {
+        serac::detail::ThreadExecutionConfiguration exec_config{.blocksize = 128};
+
+        domain_integral::gradient_kernel_cuda<geometry, test_space, trial_space, Q,
+                                              serac::detail::ThreadExecutionPolicy::THREAD_PER_QUADRATURE_POINT>(
+            exec_config, dU, dR, qf_derivatives.get(), J_, num_elements);
+      };
+#else
+      SLIC_ERROR_ROOT("serac::gpu_policy not supported on non-cuda platforms at the moment");
```

Comment:
Since this is within an `if constexpr` block can this be a compile-time error - either with https://en.cppreference.com/w/cpp/preprocessor/error or https://artificial-mind.net/blog/2020/10/03/always-false